### PR TITLE
taskjuggler: add 3.5.0, keep 2.4.3

### PR DESCRIPTION
**Pull request by khumba**
Taskjuggler is a rewrite of the 2.x series in Ruby.  I wanted this since Org mode's Taskjuggler export is for tj3 by default (although it supports both), and that what I am using over on Gentoo.  Version 3 doesn't have the Qt UI from version 2.

What's the policy for changing attribute names of packages, or for that matter, for versioned attribute (or directory) names?  Would it be better to give tj3 the 'taskjuggler' attribute name, and rename the older one, since I assume most people would want the new package?

Tested via Org's export.  It exports successfully, despite Emacs thinking it failed because of a Ruby warning that gets written to stderr (despite the exit status being 0) -- there's a duplicate key in a hash in some code, and I didn't see how to patch it out with the new Ruby builder tooling.

### CODE REVIEW

**Comment by khumba** [comment]
CC @cstrahan in case this is being or has been readded somewhere else since 8085f08 and I just haven't seen it :).

**Comment by cstrahan** [comment]
Hi Khumba! Thanks for the contribution. I'm on vacation until the 14th,
and left my laptop at home. I can take a look at this when I get back,
or someone else can feel free to give the thumbs up and merge.

File: pkgs/applications/misc/taskjuggler-3/gemset.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    ];
+  };
+  "tins" = {
+    version = "1.6.0";
+    source = {
+      type = "gem";
+      sha256 = "02qarvy17nbwvslfgqam8y6y7479cwmb1a6di9z18hzka4cf90hz";
+    };
+  };
+}
```
Comment (jagajaga) [inline pkgs/applications/misc/taskjuggler-3/gemset.nix]: Add new line please.

Comment (khumba) [inline pkgs/applications/misc/taskjuggler-3/gemset.nix]: Would you mind filing a bug with bundix for this please?  This is autogenerated and 17 of the 20 `gemfile.nix`s in the tree are missing the trailing newline.

Comment (khumba) [inline pkgs/applications/misc/taskjuggler-3/gemset.nix]: On Mon, Sep 07, 2015 at 06:54:15AM -0700, Charles Strahan wrote:
> Hi Khumba! Thanks for the contribution. I'm on vacation until the 14th,
> and left my laptop at home. I can take a look at this when I get back,
> or someone else can feel free to give the thumbs up and merge.

No rush; thanks for even replying on vacation.

Please don't merge this yet: in addition to the (harmless) warning
that tj3 emits on stderr when run, tj3man fails to run due to some
syntax errors, and I want to try with an older Ruby version.

**Comment by khumba** [comment]
Alright, I've updated the PR:

- tj3man on Ruby 2.0 works as expected.  I see also that the
  Taskjuggler website recommends that version, so I've updated the Nix
  expression correspondingly.

- I've also renamed pkgs.taskjuggler to pkgs.taskjuggler-2, and made
  pkgs.taskjuggler be v3.

Please take a look; thanks.

**Comment by khumba** [comment]
taskjuggler-2 is failing to build, but it also fails with the same
message without this patch.

**Comment by nbp** [comment]
@khumba does it work locally, have you tried to configuration nix to use a chroot?

**Comment by khumba** [comment]
@nbp, tj3 was working on a full NixOS installation when I last updated
the PR.

**Comment by rasendubi** [comment]
(triage) Tested on both NixOS and Linux (using chroot) -- builds fine after merging in master. Can we retrigger Travis and/or merge?

**Comment by bjornfor** [comment]
git rebase and force push should cause Travis to rebuild.

**Comment by khumba** [comment]
Wow, this branch is really old.  I'll rebase.

**Comment by khumba** [comment]
Rebased; haven't re-run or tested it yet.

**Comment by rasendubi** [comment]
@khumba could you look at travis failure?

**Comment by khumba** [comment]
taskjuggler-3 is fine; the failure is taskjuggler-2, which is broken already.  I wanted to preserve it because tj-3 is a complete rewrite and lacks tj-2's GUI.  I admit I'm not using tj at all now, so I'm not terribly interested in fixing tj-2.  Since that's the case, I'll drop tj-2 if you prefer.

**Comment by bjornfor** [comment]
Rebased and built locally. Pushed to master (df6827a82495c4f7f9219569a52ae3c175728111) and release-16.09 (4b7c3f234067eaf7dafa4ed3242733ace7353934).

Thanks for bringing tj3 back :-)

Btw, is taskjuggler picky about ruby version? (E.g. it doesn't work with the latest ruby?)

**Comment by khumba** [comment]
Thanks for merging!

I recall trying to use the latest Ruby back when I wrote this, and it not working, but I don't know what the issue was now.

**Pull request closed without merging**